engine/classes/ElggMemcache.php

```diff
@@ -107,6 +107,9 @@ public function __construct($namespace = 'default') {
 		if (isset($this->CONFIG->memcache_expires)) {
 			$this->expires = $this->CONFIG->memcache_expires;
 		}
+		
+		// make sure memcache is reset
+		_elgg_services()->events->registerHandler('cache:flush', 'system', array($this, 'clear'));
```

Comment:
As we here use method called `clear` I'd prefer event to be called `cache:clear` as well.